code/modules/mob/living/simple_animal/friendly/drone/interaction.dm

```diff
@@ -103,6 +108,16 @@
 	else
 		..()
 
+/mob/living/simple_animal/drone/getarmor(def_zone, type)
+	var/armorval = 0
+
+	if(istype(head, /obj/item/clothing) && islist(head.armor))
```

Comment:
You no longer need to do the istype or islist checks, all /obj have the armor list var.